### ISSUE TITLE
[BUG] Using XAML HotReload breaks compilation targeting Android

### ISSUE THREAD

**Issue by albyrock87** [microsoft/vscode-dotnettools#1247]
### Describe the Issue

I'm trying to contribute to the MAUI project https://github.com/dotnet/maui [this commit](https://github.com/dotnet/maui/commit/45a86865ffc44a501b1c4c8d6b50ee419d1696b9)
While targeting and android device or emulator and pressing `F5` to debug [this project](https://github.com/dotnet/maui/tree/main/src/Controls/samples/Controls.Sample.Profiling) I get a compilation error.

```
/Users/<myuser>/github/maui/src/Core/src/Microsoft.VisualStudio.HotReload.MobileTapInitializer/Microsoft.VisualStudio.HotReload.MobileTapInitializer.ModuleInitializerGenerator/HotReloadMobileTapInitializer.g.cs(24,40): error CS0436: The type 'HotReloadMobileTapInitializer' in 'Microsoft.VisualStudio.HotReload.MobileTapInitializer/Microsoft.VisualStudio.HotReload.MobileTapInitializer.ModuleInitializerGenerator/HotReloadMobileTapInitializer.g.cs' conflicts with the imported type 'HotReloadMobileTapInitializer' in 'Microsoft.Maui.Essentials, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. Using the type defined in 'Microsoft.VisualStudio.HotReload.MobileTapInitializer/Microsoft.VisualStudio.HotReload.MobileTapInitializer.ModuleInitializerGenerator/HotReloadMobileTapInitializer.g.cs'. [/Users/<myuser>/github/maui/src/Core/src/Core.csproj::TargetFramework=net8.0-android]
```

Notes:
- I followed the following steps to make sure solution is clean and ready
- Disabling XAML HotReload lets me compile without issues

```
git clean -xdf
dotnet tool restore && dotnet build ./Microsoft.Maui.BuildTasks.slnf
```

### Expected Behavior

Android application should compile and start with HotReload working.

### Environment Information

NET MAUI 1.1.14 (pre-release)
C# 2.35.15 (pre-release)
C# DevKit v1.8.10 (pre-release)
=> stable channel releases have the same issue

dotnet 8.0.201

VSCode 1.90
MacOS Sonoma 14.2.1 (Processor M3)
| VS Bug <a href="https://devdiv.visualstudio.com/DevDiv/_queries/edit/2113775">2113775</a>

**Comment by ziacik** [comment]
Any workaround for this? :(

**Comment by BretJohnson** [comment]
This should be fixed now, in C# extension 2.39 and later, now published to prerelease.
If after updating the C# extension (and MAUI DevKit and C# DevKit extensions) to the latest prerelease you still see this problem, please let us know.

**Comment by BretJohnson** [comment]
This should be fixed now, in C# extension 2.39 and later, now published to prerelease.
If after updating the C# extension (and MAUI DevKit and C# DevKit extensions) to the latest prerelease you still see this problem, please let us know.
